### PR TITLE
ADD: warning hub model

### PR DESCRIPTION
**Pull request by JohanWork**
# Description

Update in accordance to https://github.com/OpenAccess-AI-Collective/axolotl/pull/1202#discussion_r1493160016 . Also fixing and updating the tests etc

### CODE REVIEW

**Comment by JohanWork** [comment]
@NanoCode012  happy for you feedback here!

**Comment by winglian** [comment]
@JohanWork I'm working on a slight refactor of doing validation with Pydantic, so let's fix and merge this after that gets merged. thanks!

**Comment by JohanWork** [comment]
> @JohanWork I'm working on a slight refactor of doing validation with Pydantic, so let's fix and merge this after that gets merged. thanks!

Sure sounds great! When that is merged I can update the pr as well!

**Comment by winglian** [comment]
@JohanWork the pydantic refactor has been merged. let me know if you have any questions about that.

File: src/axolotl/utils/config.py
Comment (NanoCode012) [inline src/axolotl/utils/config.py]: I think this check was made to ensure that one of save_steps/saves_per_epoch/save_strategy was set. In this case, you're missing those conditions?

File: tests/test_validation.py
Comment (NanoCode012) [inline tests/test_validation.py]: This test is wrong, I believe. It should warn as there's no `save_steps` or `saves_per_epoch` or `save_strategy`

File: src/axolotl/utils/config.py
Comment (JohanWork) [reply src/axolotl/utils/config.py]: Isen't the only thing that matters that you have set a save strategy? https://huggingface.co/docs/transformers/main_classes/trainer#transformers.TrainingArguments.save_strategy. And to not save you have to set it to NO as I have understood it, if you don't set it (None in axolotl) it becomes steps.

File: tests/test_validation.py
Comment (JohanWork) [reply tests/test_validation.py]: Not setting save_strategy means it becomes steps as I understand it @NanoCode012 and then the test should pass right?

File: src/axolotl/utils/config.py
Comment (NanoCode012) [reply src/axolotl/utils/config.py]: Should this be changed to check for "no" then?

File: tests/test_validation.py
Comment (NanoCode012) [reply tests/test_validation.py]: You're right.

File: src/axolotl/utils/config.py
Comment (JohanWork) [reply src/axolotl/utils/config.py]: It could be, but currently the test checks if it is no or any other value. I wanted to check more widely, but can tighten it up if you think it is better @NanoCode012 . Also sorry for slow progress :(

**Review by NanoCode012 — APPROVED** [review]
Following discussion, this seems good then!

**Comment by NanoCode012** [comment]
Hey @JohanWork , would you be able to rebase or resolve merge conflicts? I will merge this PR after.

**Comment by JohanWork** [comment]
@NanoCode012 I have update the pr. Sorry for really slow response from my side. I had to update src/axolotl/utils/config/models/input/v0_4_1/__init__.py, as previous mention setting save_steps or save_epochs dosen't matter what matters is setting save_stragey(https://huggingface.co/docs/transformers/main_classes/trainer#transformers.TrainingArguments.save_strategy) and None in axolotl defaults to steps. let me know what you think and if I have missed something. Will be faster in responding :)

**Comment by NanoCode012** [comment]
Thank you for the PR. Sorry that it took a while.

**Comment by JohanWork** [comment]
No worries :)  Thnx

**Pull request merged**
